test/math_unit_test.hpp

```diff
@@ -29,12 +30,12 @@ namespace detail {
     {
         int status = 0;
         std::size_t size = 0;
-        std::string s = name;
 #if BOOST_MATH_HAS_CXX_ABI
-        return abi::__cxa_demangle( name, NULL, &size, &status );
+        std::string s {abi::__cxa_demangle( name, NULL, &size, &status )};
 #else
-        return name;
+        std::string s {name};
```

Comment:
Apparently there is a demangler on Windows that would make this a bit cleaner. Do you have a Windows box?

Reply:
I don't right now because I am about to move back across the pond. I am going to revert back to `boost::core::demangle` because I am still getting a memory leak on the GCC CI test that I can't replicate with clang.

Reply:
Oh yeah . . . I thought std::string taking ownership of malloc'd char* was standardized now, but I guess not . . .